src/platform/unix/mod.rs

```diff
@@ -649,10 +646,64 @@ fn make_socket_lingering(sockfd: c_int) -> Result<(),UnixError> {
     Ok(())
 }
 
+struct BackingStore {
+    fd: c_int
+}
+
+impl BackingStore {
+    pub unsafe fn create(length: usize) -> BackingStore {
+        let string = create_filename();
+        let fd = create_shmem(string.as_ptr(), length);
+        BackingStore {
+            fd: fd
+        }
```

Comment:
Could use `Self::from_fd()` I guess? Not sure whether that's indeed more readable though...